### ISSUE TITLE
HashKeyBag on Int crashes GAP

### ISSUE THREAD

**Issue by markuspf** [gap-system/gap#817]
Please use the following template to submit an issue
(you may delete lines which are not used). Thank You!
### Observed behaviour

```
gap> GG := 4;;
gap> HashKeyBag(GG,1,1,-1);
[1]    3779 segmentation fault  sh bin/gap.sh
```
### Expected behaviour

Ideally getting a hash, or an error message. In the case of error message I'd want a way to tell whether I can hash this object. Arguably the only exception should be (small) ints?
### Copy and paste GAP banner (to tell us about your setup)

```
 ┌───────┐   GAP v4.8.3-612-gcd4afab of 2016-05-02 11:18:31 (BST)
 │  GAP  │   http://www.gap-system.org
 └───────┘   Architecture: x86_64-unknown-dragonfly4.5-gcc-default64
```

**Comment by fingolfin** [comment]
For a small int, I'd imagine the hash would be the small int itself.

Related:
```
gap> GG:=Z(2);
Z(2)^0
gap> HashKeyBag(GG,1,1,-1);
Segmentation fault: 11
```
However, for small FFEs, doing proper hashing is difficult by the fact that different FFEs can describe equal objects (by virtue of the automatic embedding of GF(q) into GF(q^n) performed by gap). I.e.:
```
gap> Z(2) = Z(2^4)^0;
true
gap> IsIdenticalObj(Z(2), Z(2^4)^0);
false
```
Of course one could always "normalize" FFEs before computing their hash, but that comes at a performance penalty.

**Comment by fingolfin** [comment]
I added a fix for this on PR #1048. It is somewhat arbitrary in that it changes `HashKeyBag` to handle immediate ints given as input object by returning the input object as output, and refusing to handle FFEs. Note that the "arbitrary" bit is to accept immediate integers; a case could be made that those should be rejected, too. If somebody wants it differently, speak up. In the meantime, either behaviour is better than a crash :)

**Comment by fingolfin** [comment]
Fixed in 3306f0fbfc9033603a95d75412e79332083aa42b
